### PR TITLE
[torch_glow] Add support for sum op

### PR DESCRIPTION
**Pull request by sgeorgiev255**
Summary:
Add support for the sum op in PyTorchModel loader.

Test Plan:
Relevant unit tests have been added to sum_test.py.

### CODE REVIEW

**Review by 842974287 — COMMENTED** [review]
@sgeorgiev255 Thanks for adding this. Looks great! In `loadSum` we could check the size of inputs and load the inputs accordingly. IIRC for aten::sum there would be either 2 or 4 inputs.

File: torch_glow/tests/nodes/sum_test.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+class KeepdimSumModule(torch.nn.Module):
+    def __init__(self, axis, keepdim, dtype=torch.float32):
+        super(KeepdimSumModule, self).__init__()
+        self.axis = axis
+        self.keepdim = keepdim
+        self.dtype = dtype
+
+    def forward(self, a):
+        b = a + a
+        return torch.sum(a)
```
[the remark this answers is not in the document]
Comment (sgeorgiev255) [reply torch_glow/tests/nodes/sum_test.py]: My bad, it should be, fixed it.

File: torch_glow/tests/nodes/sum_test.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import unittest
+
+import torch
+from parameterized import parameterized
+from tests import utils
+
+
+class SimpleSumModule(torch.nn.Module):
+    def __init__(self, dtype=torch.float32):
```
[the remark this answers is not in the document]
Comment (sgeorgiev255) [reply torch_glow/tests/nodes/sum_test.py]: Fixed.

File: torch_glow/tests/nodes/sum_test.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+class KeepdimSumModule(torch.nn.Module):
+    def __init__(self, axis, keepdim, dtype=torch.float32):
+        super(KeepdimSumModule, self).__init__()
+        self.axis = axis
+        self.keepdim = keepdim
+        self.dtype = dtype
+
+    def forward(self, a):
+        b = a + a
```
[the remark this answers is not in the document]
Comment (sgeorgiev255) [reply torch_glow/tests/nodes/sum_test.py:28]: Fixed.

File: torch_glow/tests/nodes/sum_test.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        super(SimpleSumModule, self).__init__()
+        self.dtype = dtype
+
+    def forward(self, a):
+        b = a + a
+        return torch.sum(a, dtype=self.dtype)
+
+
+class KeepdimSumModule(torch.nn.Module):
+    def __init__(self, axis, keepdim, dtype=torch.float32):
```
[the remark this answers is not in the document]
Comment (sgeorgiev255) [reply torch_glow/tests/nodes/sum_test.py]: Good point, fixed.

**Comment by sgeorgiev255** [comment]
> @sgeorgiev255 Thanks for adding this. Looks great! In `loadSum` we could check the size of inputs and load the inputs accordingly. IIRC for aten::sum there would be either 2 or 4 inputs.

Thanks for the comments @842974287. I have refactored the `loadSum` function based on your suggestions.

**Comment by sgeorgiev255** [comment]
> > @sgeorgiev255 Thanks for adding this. Looks great! In `loadSum` we could check the size of inputs and load the inputs accordingly. IIRC for aten::sum there would be either 2 or 4 inputs.
> 
> Thanks for the comments @842974287. I have refactored the `loadSum` function based on your suggestions.

Hi @842974287, any feedback on the changes?

**Comment by 842974287** [comment]
@sgeorgiev255  Sorry for the delay, I'll take a look later today!

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 5]
```diff
@@ -2306,6 +2307,93 @@ Error PyTorchModelLoader::loadLog(const torch::jit::Node *ptNode) {
...
+        "flatten", needsConvertTo ? static_cast<NodeValue>(toNode) : input,
+        needsConvertTo ? toNode->getResult().getType()->dims().size()
+                       : input.dims().size());
+  }
+
+  auto batchedReduceAddNode = F_.createBatchedReduceAdd(
+      "sum",
+      needsFlatten ? static_cast<NodeValue>(flattenNode)
+                   : needsConvertTo ? static_cast<NodeValue>(toNode) : input,
+      glowAxes);
```
Comment (842974287) [inline torch_glow/src/PyTorchModelLoader.cpp:2379]: BatchedReduceAddNode doesn't support multi-axis. Might just return error when seeing multi-axis. https://github.com/pytorch/glow/blob/master/lib/Graph/Graph.cpp#L1888

**Review by 842974287 — COMMENTED** [review]
LG! Just one comment.

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 5 above)
Comment (sgeorgiev255) [reply torch_glow/src/PyTorchModelLoader.cpp:2491]: Done.

**Review by 842974287 — APPROVED** [review]
